test/metabase/query_processor/middleware/auto_parse_filter_values_test.clj

```diff
@@ -26,21 +28,32 @@
                                 :type/Boolean    true}]
     (testing (format "A String parameter that has %s should get parsed to a %s"
                      base-type (.getCanonicalName (class expected)))
-      (is (= (lib.tu.macros/$ids venues
-               [:= $price [:value expected {:base_type base-type}]])
-             (-> (lib.tu.macros/mbql-query venues
-                   {:filter [:= $price [:value (str expected) {:base_type base-type}]]})
-                 auto-parse-filter-values
-                 :query :filter))))))
+      (let [query (-> (lib/query meta/metadata-provider (meta/table-metadata :venues))
+                      (lib/filter (lib/= (meta/field-metadata :venues :price)
+                                         ;; apparently we have no MLv2 helper for creating `:value` clauses
+                                         [:value
+                                          {:base-type base-type, :effective-type base-type, :lib/uuid (str (random-uuid))}
+                                          (str expected)])))]
```

Comment:
Not strictly necessary to rework this test to use lib functions to create the query but I wanted to demo how to do it.